hermes-api/src/main/java/pl/allegro/tech/hermes/api/TrackingMode.java

```diff
@@ -2,7 +2,41 @@
 
 public enum TrackingMode {
 
-    TRACK_ALL,
-    TRACK_DISCARDED_ONLY,
-    TRACKING_OFF
+    TRACK_ALL("trackingAll"),
+    TRACK_DISCARDED_ONLY("discardedOnly"),
+    TRACKING_OFF("trackingOff");
+
+    private String value;
+
+    TrackingMode(String s) {
+        this.value = s;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Override
+    public String toString() {
+        return this.value;
+    }
+
+    public static TrackingMode fromString(String trackingMode) {
+
+        if(trackingMode == null) {
```

Comment:
I don't like the returning null part, how about an `Optional<TrackingMode>`?
And then:
```java
TrackingMode validTrackingMode = TrackingMode.fromString(trackingMode).orElse(trackingEnabled ? TrackingMode.TRACK_ALL : TrackingMode.TRACKING_OFF);
```

Reply:
Done :)